lib/utils.js

```diff
@@ -493,8 +493,12 @@ function createPromiseCallback() {
 
   var promise = new Promise(function(resolve, reject) {
     cb = function(err, data) {
-      if (err) return reject(err);
-      return resolve(data);
+      if (err) {
+        reject(err);
+      } else {
+        resolve(data);
+      }
+      return promise;
```

Comment:
This change is not fixing the problem. The return value of a callback is almost always discarded, therefore it does not matter what this function is returning.